### PR TITLE
Spawn reactive runner

### PR DESCRIPTION
**Pull request by cbartz**
Applicable spec: [ISD-116](https://discourse.charmhub.io/t/specification-isd116-github-self-hosted-runners-reactive-scheduling/13755)

### Overview

Spawn a reactive runner after consuming a message from the queue. See https://github.com/canonical/github-runner-manager/pull/8

This PR adapts the code to the changes in https://github.com/canonical/github-runner-manager/pull/8. It also

- Adapts the reactive integration test to check that a reactive runner has been spawned.
- Removes reactive support for LXD, the charm will now go into blocked state if mongodb is integrated without Openstack configured.
- Configures logrotate properly for Openstack.

### Rationale

See rationale in https://github.com/canonical/github-runner-manager/pull/8 .

logrotate working correctly is important as the number of log files increases with reactive mode.

### Juju Events Changes

n/a

### Module Changes

- `charm`: Adjustments to use modified ReactiveConfig.
- `charm_state`: Raise error if LXD and Reactive Mode are used simultaneously.

### Library Changes

n/a

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The changelog is updated with changes that affects the users of the charm.

[ISD-116]: https://warthogs.atlassian.net/browse/ISD-116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -12,6 +12,4 @@ cosl ==0.0.15
 # juju 3.1.2.0 depends on pyyaml<=6.0 and >=5.1.2
 PyYAML ==6.0.*
 pyOpenSSL==24.2.1
-kombu==5.4.1
-pymongo==4.8.0
-github_runner_manager @ git+https://github.com/canonical/github-runner-manager.git@6ee40e296f92f191dd12b1e24a613dfdbdc70f99
+github_runner_manager @ git+https://github.com/canonical/github-runner-manager.git@feat/reactive-spawning
```
Comment (jdkandersson) [inline requirements.txt:15]: We'll need to fix this before merging

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by amandahla — DISMISSED** [review]

File: src/charm.py [hunk 2]
```diff
@@ -1293,25 +1335,16 @@ def _get_runner_scaler(
...
-            token=token,
-            path=path,
-        )
-        runner_manager = RunnerManager(
-            manager_name=self.app.name,
-            cloud_runner_manager=openstack_runner_manager,
-            config=runner_manager_config,
-        )
-        return RunnerScaler(runner_manager=runner_manager, reactive_config=state.reactive_config)
+        return openstack_runner_manager_config
```
Comment (yanksyoon) [inline src/charm.py:1347]: Should this return a runner scaler type? it seems like it is returning a configuration.

**Review by yanksyoon — DISMISSED** [review]
Minor question, LGTM, Thanks!

**Comment by yanksyoon** [comment]
Resolved merge conflict to use latest runner-manager lib.

**Review by jdkandersson — DISMISSED** [review]

**Review by yanksyoon — DISMISSED** [review]

File: src/charm.py [hunk 3]
```diff
@@ -448,6 +451,12 @@ def _common_install_code(self, state: CharmState) -> bool:  # noqa: C901
             logger.error("Failed to install charm dependencies")
             raise
 
+        try:
+            logrotate.setup()
+        except LogrotateSetupError:
+            logger.error("Failed to setup logrotate")
+            raise
+
```
Comment (cbartz) [inline src/charm.py:459]: Move logrotate up to also rotate logs for Openstack

File: requirements.txt [hunk 4]
```diff
@@ -12,6 +12,4 @@ cosl ==0.0.15
 # juju 3.1.2.0 depends on pyyaml<=6.0 and >=5.1.2
 PyYAML ==6.0.*
 pyOpenSSL==24.2.1
-kombu==5.4.1
-pymongo==4.8.0
-github_runner_manager @ git+https://github.com/canonical/github-runner-manager.git@59974da80e06907c3e1650d587ef0986f3b6db6d
+github_runner_manager @ git+https://github.com/canonical/github-runner-manager.git@5789edf2f62bdebf7151f375875b540263d3605c
```
Comment (cbartz) [inline requirements.txt:15]: Latest commit includes url key error fix: https://github.com/canonical/github-runner-manager/pull/16

**Review by yanksyoon — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**